src/iop/hotpixels.c

```diff
@@ -203,81 +265,28 @@ static int process_xtrans(const void *const ivoid, void *const ovoid,
 {
   dt_iop_hotpixels_gui_data_t *g = (dt_iop_hotpixels_gui_data_t *)self->gui_data;
   const dt_iop_hotpixels_data_t *data = (dt_iop_hotpixels_data_t *)piece->data;
-  const float threshold = data->threshold;
-  const float multiplier = data->multiplier;
-  const int width = roi_out->width;
-  const int height = roi_out->height;
-  const int widthx2 = width * 2;
-  const gboolean markfixed = data->markfixed;
   const int min_neighbours = data->permissive ? 3 : 4;
 
-  // The loop should output only a few pixels, so just copy everything first
-  memcpy(ovoid, ivoid, (size_t)width * height * sizeof(float));
-
-  int fixed = 0;
+  // The processing loop should output only a few pixels, so just copy everything first
+  memcpy(ovoid, ivoid, (size_t)roi_out->width * roi_out->height * sizeof(float));
 
+  int fixed;
   if(piece->pipe->filters == 9u)
   {
     fixed = process_xtrans(ivoid, ovoid, roi_out, (const uint8_t(*const)[6])piece->pipe->xtrans,
-                           threshold, multiplier, markfixed, min_neighbours);
-    goto processed;
+                           data->threshold, data->multiplier, data->markfixed, min_neighbours);
   }
-
-// Bayer sensor array
-#ifdef _OPENMP
-#pragma omp parallel for default(none) reduction(+ : fixed) schedule(static)
-#endif
-  for(int row = 2; row < roi_out->height - 2; row++)
+  else
   {
-    const float *in = (float *)ivoid + (size_t)width * row + 2;
-    float *out = (float *)ovoid + (size_t)width * row + 2;
-    for(int col = 2; col < width - 1; col++, in++, out++)
-    {
-      float mid = *in * multiplier;
-      if(*in > threshold)
-      {
-        int count = 0;
-        float maxin = 0.0;
-        float other;
-#define TESTONE(OFFSET)                                                                                      \
-  other = in[OFFSET];                                                                                        \
-  if(mid > other)                                                                                            \
-  {                                                                                                          \
-    count++;                                                                                                 \
-    if(other > maxin) maxin = other;                                                                         \
-  }
-        TESTONE(-2);
-        TESTONE(-widthx2);
-        TESTONE(+2);
-        TESTONE(+widthx2);
-#undef TESTONE
-        if(count >= min_neighbours)
-        {
-          *out = maxin;
-          fixed++;
-          if(markfixed)
-          {
-            for(int i = -2; i >= -10 && i >= -col; i -= 2) out[i] = *in;
-            for(int i = 2; i <= 10 && i < width - col; i += 2) out[i] = *in;
-          }
-        }
-      }
-    }
+    fixed = process_bayer(ivoid, ovoid, roi_out,
+                          data->threshold, data->multiplier, data->markfixed, min_neighbours);
```

Comment:
Thank you! Made a commit to do this. I also eliminated passing `min_neighbors,` as this is derived from `dt_iop_hotpixels_data_t*`, at the cost of building that logic into the `process_*()` functions.